types/Link.d.ts

```diff
@@ -0,0 +1,25 @@
+import 'svelte2tsx/svelte-jsx'
+import { SvelteComponentTyped } from 'svelte'
+import { RouteLocation } from './Route'
+
+interface LinkProps {
+  to: string
+  replace?: boolean
+  state?: {
+    [k in string | number]: unknown
+  }
+  getProps?: (linkParams: GetPropsParams) => Record<string, any>
+}
+
+interface GetPropsParams {
+  location: RouteLocation
+  href: string
+  isPartiallyCurrent: boolean
+  isCurrent: boolean
+}
+
+type Link = SvelteComponentTyped<
+  Omit<LinkProps & svelte.JSX.HTMLProps<HTMLAnchorElement> & svelte.JSX.SapperAnchorProps, 'href'>
+>
```

Comment:
```suggestion
  class Link extends SvelteComponentTyped<
    Omit<LinkProps & svelte.JSX.HTMLProps<HTMLAnchorElement> & svelte.JSX.SapperAnchorProps, 'href'>
  > {}
```

Reply:
Just out of curiosity, why it should be a `class` instead a `type`?

Reply:
Same as https://github.com/EmilTholin/svelte-routing/pull/117#discussion_r564267515